lib/pages/chat/chat_invitation_body.dart

```diff
@@ -0,0 +1,282 @@
+import 'package:fluffychat/pages/chat/chat.dart';
+import 'package:fluffychat/pages/chat/chat_invitation_body_style.dart';
+import 'package:fluffychat/pages/chat/dialog_reject_invite_widget.dart';
+import 'package:fluffychat/pages/chat/events/message_content_mixin.dart';
+import 'package:fluffychat/utils/dialog/twake_dialog.dart';
+import 'package:fluffychat/widgets/matrix.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:linagora_design_flutter/linagora_design_flutter.dart';
+
+class ChatInvitationBody extends StatelessWidget with MessageContentMixin {
+  final ChatController controller;
+
+  const ChatInvitationBody(this.controller, {Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    controller.matrix ??= Matrix.of(context);
+    final client = controller.matrix!.client;
+    controller.sendingClient ??= client;
+    controller.room = controller.sendingClient!.getRoomById(controller.roomId!);

```

Comment:
Put what you need in the constructor, don't put the whole controller in constructor, it's hard to reuse

Reply:
I agree with @sherlockvn, create some properties for ti

Reply:
Totally agree, do any logic in `build()` is not a smart choice